Actions/CreateApp/AppHelper.psm1

```diff
@@ -0,0 +1,136 @@
+<#
+This module contains some useful functions for working with app manifests.
+#>
+
+. (Join-Path -path $PSScriptRoot -ChildPath "..\AL-Go-Helper.ps1" -Resolve)
+$here = Split-Path -Parent $MyInvocation.MyCommand.Path
+$alTemplatePath = Join-Path -Path $here -ChildPath "AppTemplate" 
+
+
+$validRanges = @{
+    "PTE"           = "50000..99999";
+    "AppSource App" = "100000..$([int32]::MaxValue)";
+    "Test App"      = "50000..$([int32]::MaxValue)" ;
+};
+
+function Confirm-IdRanges([string] $templateType, [string]$idrange ) {  
+    $validRange = $validRanges.$templateType.Replace('..', '-').Split("-")
+    $validStart = [int](stringToInt($validRange[0]))
+    $validEnd = [int](stringToInt($validRange[1]))
+
+    $ids = $idrange.Replace('..', '-').Split("-")
+    $idStart = [int](stringToInt($ids[0]))
+    $idEnd = [int](stringToInt($ids[1]))
+    
+    if ($ids.Count -ne 2 -or ($idStart) -lt $validStart -or $idStart -gt $idEnd -or $idEnd -lt $validStart -or $idEnd -gt $validEnd -or $idStart -gt $idEnd) { 
+        throw "IdRange should be formattet as fromId..toId, and the Id range must be in $($validRange[0]) and $($validRange[1])"
+    }
+
+    return $ids
+} 
+
+function UpdateManifest
+(
+    [string]$appJsonFile,
```

Comment:
A good idea is to have a space between the type and the variable.
In PS IDE, when you press Ctrl+F on a variable declaration, it will then pick only the variable to search for